File: py-proxy-tester-project-authenticated/auth-sessions/create.py
from typing import TypedDict

from intuned_browser import go_to_url
from playwright.async_api import Page

from utils.egress import assert_expected_egress_ip


class Params(TypedDict, total=False):
    username: str
    password: str
    expectedEgressIp: str
    egressProbeUrl: str


async def create(page: Page, params: Params | None = None, **_kwargs):
    if params is None:
        raise ValueError("Params with username and password are required")

    await go_to_url(page=page, url="https://sandbox.intuned.dev/login")

    email_input = page.locator("#email-input")
    await email_input.fill(params["username"])

    password_input = page.locator("#password-input")
    await password_input.fill(params["password"])

    submit_button = page.locator("#submit-button")
    await submit_button.click()

    protected_page = page.locator("#book-consultations-title")
    await protected_page.wait_for(state="visible", timeout=10_000)

    # Only treat the session as created once it is confirmed to have been built
    # through the intended proxy. Creating it through the wrong egress would
    # otherwise produce a session that quietly fails its own checks later.
    await assert_expected_egress_ip(page, dict(params), "AuthSession create")

File: ts-proxy-tester-project-authenticated/auth-sessions/create.ts
import { BrowserContext, Page } from "playwright";
import { goToUrl } from "@intuned/browser";

import { assertExpectedEgressIp, EgressExpectation } from "../utils/egress";

export interface CreateAuthSessionParams extends EgressExpectation {
  username: string;
  password: string;
}

export default async function create(
  params: CreateAuthSessionParams,
  page: Page,
  _context: BrowserContext
): Promise<void> {
  await goToUrl({ page, url: "https://sandbox.intuned.dev/login" });

  const emailInput = page.locator("#email-input");
  await emailInput.fill(params.username);

  const passwordInput = page.locator("#password-input");
  await passwordInput.fill(params.password);

  const submitButton = page.locator("#submit-button");
  await submitButton.click();

  const protectedPage = page.locator("#book-consultations-title");
  await protectedPage.waitFor({ state: "visible", timeout: 10_000 });

  // Only treat the session as created once it is confirmed to have been built
  // through the intended proxy. Creating it through the wrong egress would
  // otherwise produce a session that quietly fails its own checks later.
  await assertExpectedEgressIp(page, params, "AuthSession create");
}
